### ISSUE TITLE
Double entries of media files in database - one per station for the same media file

### ISSUE THREAD

**Issue by sterreke** [AzuraCast/AzuraCast#636]
**Is your feature request related to a problem? Please describe.**
I have two stations configured. Both stations point to the exact same media folder (in my case a nfs share).
Why does azuracast need to process the same media folder for each station? If a media file is processed for one station, shouldn't that be enough?
I have a very large media collection (+50k) media files. And now I have a Azuracast database with +100k media file database entries. This is a waste of diskspace and processing resources.

Example in my database:

MariaDB [azuracast]> select * from station_media where song_id='0cd724fc7104a9a44e90cfccea95b47a' limit 5;
+-------+------------+----------------------------------+---------------------+--------+------------+--------+-------------+----------------------------------------------------------------------------------+------------+--------------+---------+----------+--------+---------+------+--------+--------------------------+
| id    | station_id | song_id                          | title               | artist | album      | length | length_text | path                                                                             | mtime      | fade_overlap | fade_in | fade_out | cue_in | cue_out | isrc | lyrics | unique_id                |
+-------+------------+----------------------------------+---------------------+--------+------------+--------+-------------+----------------------------------------------------------------------------------+------------+--------------+---------+----------+--------+---------+------+--------+--------------------------+
|  9151 |          1 | 0cd724fc7104a9a44e90cfccea95b47a | Squeeze The Trigger | Ice?T  | Rhyme Pays |    375 | 6:14        | Flac/Hip-Hop/Ice?T/Ice?T - 1987 - Rhyme Pays/09 Squeeze The Trigger - Ice?T.flac | 1511049593 | NULL         | NULL    | NULL     | NULL   | NULL    | NULL | NULL   | 129b1ec26fe4e1e13bafb1b7 |
| 45566 |          2 | 0cd724fc7104a9a44e90cfccea95b47a | Squeeze The Trigger | Ice?T  | Rhyme Pays |    375 | 6:14        | Flac/Hip-Hop/Ice?T/Ice?T - 1987 - Rhyme Pays/09 Squeeze The Trigger - Ice?T.flac | 1511049593 | NULL         | NULL    | NULL     | NULL   | NULL    | NULL | NULL   | 8098d45c83d5a4f8a6664b2c |
+-------+------------+----------------------------------+---------------------+--------+------------+--------+-------------+----------------------------------------------------------------------------------+------------+--------------+---------+----------+--------+---------+------+--------+--------------------------+

**Describe the solution you'd like**
One entry per media file in the database, shared among stations. Media file should only be processed once.

**Describe alternatives you've considered**
N/A/

**Additional context**
N/A.

**Comment by SlvrEagle23** [comment]
@sterreke The way the relational database is set up in this case means that the optimal solution for one song being processed on two different stations is to list it twice in the database.

The short explanation is that there are a number of tables that all relate to the `station_media` table, and they're all referring to that media item on _one given station_. Playback history, playlist membership, etc. all depend on that media record referring to one media file on one station.

While this may occupy a lot of space for very large scale stations that operate with tens of thousands of records, changing the fundamental structure of the database at this level would be a significant undertaking that would be very, very error prone. 

It's worth noting (as I often do) that this project is almost entirely the sole effort of one single developer (me) working on it as a labor of love while trying (and lately failing) to keep a roof over my head by completing other paying work at the same time. A change of this magnitude that affects a large amount of how AzuraCast processes and controls media would take a very large amount of time and would only be of marginal benefit to station operators working at a scale far beyond what I can test for locally.

In other words, I understand the concern here and it's valid, but it may be beyond the scope of my time and ability as a volunteer to make the needed changes in a way that will gracefully migrate all existing users with no data loss. I'll leave the item open and keep it for future consideration.

**Comment by sterreke** [comment]
Thank you for considering this.

**Comment by SlvrEagle23** [comment]
At this point, I believe there's too much tied into the current infrastructure to consider a change of this magnitude. Particularly, combining the StationMedia entity entries for multiple stations would allow one station's administrators to control the metadata of another station's media even if permissions wouldn't allow it.

I will continue to work on optimizations, particularly with station cloning, so the system doesn't encounter performance problems.
